gamemodes/terrortown/gamemode/server/sv_shop.lua

```diff
@@ -221,9 +223,31 @@ local function TransferCredits(ply, cmd, args)
 
 	if credits == 0 then return end
 
+	---
+	-- Called to check if a transaction between two players is allowed.
+	-- @param Player that wants to send credits
+	-- @param Player that would receive the credits
+	-- @param Number of credits that would be transferred
+	-- @return[default=nil] Boolean which disallows a transaction when false
+	-- @return[default=nil] String for the client which offers info related to the transaction
+	-- @realm server
+	local allow, _ = hook.Run("TTT2CanTransferCredits", ply, target, credits)
+	if allow == false then return end
+
 	ply:SubtractCredits(credits)
-	target:AddCredits(credits)
+	if target:Alive() then
+		target:AddCredits(credits)
+	else
+		-- The would be recipient is dead, which the sender may not know.
+		-- Instead attempt to send the credits to the target's corpse, where they can be picked up.
+		local rag = target:FindCorpse()
+		if IsValid(rag) then
+			CORPSE.SetCredits(rag, CORPSE.GetCredits(rag, 0) + credits)
+		end
```

Comment:
Maybe we should discuss about what happens if the corpse was burned as well or got destroyed somehow else. Should we display a message as well about the lost credits? That would e.g. add a new mechanic to check for alive players with spending credits. But yea, that could break many things. So are we sure we just wanna let those credits get lost?

Reply:
I'd say to just let them be lost. So we do not introduce a new mechanic (which could also still be done by an add-on)